pkg/controller/atlas/client.go

```diff
@@ -13,13 +13,19 @@ import (
 	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/httputil"
 )
 
+type HTTPClientFn func() *http.Client
+
+// CustomHTTPClientFn allows to override the default basic http.Client,
+// for protocol logging or test mocking
+var CustomHTTPClientFn HTTPClientFn
```

Comment:
I'm not in favor of adding a global as a way to override a dependency. What do you think about user Functional Options to configure the client?
I understand it will be more work but I believe it's a better approach.

Reply:
That is cleaner indeed, I will refactor.

Reply:
Done